js/util.js

```diff
@@ -5,10 +5,48 @@ app.util = (function() {
 	'use strict';
 
 	return {
+
 		extend: function() {
+
 			var args = Array.prototype.slice.call(arguments);
 			return _.extend.apply(_, [{}].concat(args));
+		},
+
+		generateRandomString: function(length, charset) {
+
+			var randString = '';
+
+			charset = charset || 'abcdefghijklmnopqrstuvqxyzABCDEFGHIJKLMNOPQRSTUVQXYZ1234567890';
+
+			var randomValues = app.util.getRandomValues(length);
+
+			for (var index = 0; index < length; index++) {
+				randString += charset[randomValues[index] % charset.length];
+			}
+
+			return randString;
+		},
+
+		getRandomValues: function(length) {
+
+			var randomValues;
+
+			// This should work for the latest browsers:
+			if (window.crypto && window.crypto.getRandomValues) {
+				randomValues = new Uint32Array(length);
+				window.crypto.getRandomValues(randomValues);
+			// Fallback for IE:
+			} else if (window.msCrypto && window.msCrypto.getRandomValues) {
+				randomValues = new Int32Array(length);
+				window.msCrypto.getRandomValues(randomValues);
+			// And if none of the above works, try using the SJCL library:
+			} else {
+				randomValues = sjcl.random.randomWords(length);
```

Comment:
Are we adding `sjcl` as a dependency just for this fallback?

Reply:
I think we will need sjcl elsewhere as well.